### DISCUSSION TITLE
How to create theorems without giving it an id?

### DISCUSSION THREAD

**Discussion by chaoxu** [quarto-dev/quarto-cli#3499] [Q&A] [answered]
I see it is possible to create theorems as follows.

```
::: {#thm-id-of-the-theorem} 
# theorem title
theorem body
:::
```

But then I need to give a id to every theorem. Often it will not be referred anywhere else, and I try to not give things id when it is unnecessary(for example, it pollute the dropdown when i want to select what to refer). Is there a way to create a theorem without giving it a name?

**Comment by cscheid** [comment, marked as answer]
Not currently, because we use the id to identify the div as a theorem. You could write a lua filter that detects the div you want to be a theorem (by whatever means you choose) and then give it an auto-generated id.
